packages/amplify-graphql-index-transformer/src/graphql-index-transformer.ts

```diff
@@ -88,6 +88,7 @@ export class IndexTransformer extends TransformerPluginBase {
 
 function validate(config: IndexDirectiveConfiguration, ctx: TransformerContextProvider): void {
   const { name, object, field, sortKeyFields } = config;
+  const defaultGSI = ctx.featureFlags.getBoolean('secondaryKeyAsGSI', false);
```

Comment:
I think this should have a default value of `true`.

```suggestion
  const defaultGSI = ctx.featureFlags.getBoolean('secondaryKeyAsGSI', true);
```